src/components/multi-action-button/multi-action-button.pw.tsx

```diff
@@ -0,0 +1,946 @@
+      await expect(listButton3).toBeFocused();
+    });
+
+    test(`should verify that pressing Esc key closes MultiActionButton`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<WithWrapper />);
+
+      const actionButtonComp = getComponent(
+        page,
+        "multi-action-button"
+      ).first();
+      await actionButtonComp.click();
+      await expect(
+        getDataElementByValue(page, "additional-buttons")
+      ).toHaveCount(1);
+      await page.keyboard.press("Escape");
+      await expect(
+        getDataElementByValue(page, "additional-buttons")
+      ).toHaveCount(0);
+    });
+
+    test(`should verify that clicking one of the child buttons closes MultiActionButton`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<WithWrapper />);
+
+      await getComponent(page, "multi-action-button").first().click();
+      await expect(
+        getDataElementByValue(page, "additional-buttons")
+      ).toHaveCount(1);
+      await page.getByRole("menuitem").nth(0).click();
+      await expect(
+        getDataElementByValue(page, "additional-buttons")
+      ).toHaveCount(0);
+    });
+  }
+);
+
+test.describe("Accessibility tests", () => {
+  test(`should pass tests for default component`, async ({ mount, page }) => {
+    await mount(<MultiActionButtonList />);
+
+    await checkAccessibility(page);
+  });
+
+  // TODO: test passes even when it shouldn't, see FE-6267
+  test(`should pass tests when open`, async ({ mount, page }) => {
+    await mount(<MultiActionButtonList />);
+
+    await page.getByRole("button").focus();
+    await page.keyboard.press("Enter");
+
+    await checkAccessibility(page);
+  });
+
+  test(`should pass tests with disabled prop set`, async ({ mount, page }) => {
+    await mount(<MultiActionButtonList disabled />);
+
+    await checkAccessibility(page);
+  });
+
+  ([
+    SIZE.SMALL,
+    SIZE.MEDIUM,
+    SIZE.LARGE,
+  ] as MultiActionButtonProps["size"][]).forEach((size) => {
+    test(`should pass tests with size prop set to ${size}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<MultiActionButtonList size={size} />);
+
+      await checkAccessibility(page);
+    });
+  });
+
+  test(`should pass tests with width prop set`, async ({ mount, page }) => {
+    await mount(<MultiActionButtonList width="70%" />);
+
+    await checkAccessibility(page);
+  });
+
+  ([
+    "primary",
+    "secondary",
+    "tertiary",
+  ] as MultiActionButtonProps["buttonType"][]).forEach((type) => {
+    test(`should pass tests with buttonType prop set as ${type}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<MultiActionButtonList buttonType={type} />);
+
+      await checkAccessibility(page);
+    });
+  });
+
+  test(`should pass tests for ChildButtonTypes example`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<ChildButtonTypes />);
+
+    await checkAccessibility(page);
+  });
```

Comment:
**suggestion(non-blocking):** You will probably need to have the `additional-buttons` menu rendered here for this accessibility check to mean anything.